pallets/loans/src/entities/pricing/external.rs

```diff
@@ -148,10 +158,36 @@ impl<T: Config> ExternalActivePricing<T> {
 		Ok(self.outstanding_quantity.ensure_mul_int(price)?)
 	}
 
+	fn validate_amount(
+		&self,
+		amount: &ExternalAmount<T>,
+		pool_id: T::PoolId,
+	) -> Result<(), DispatchError> {
+		let price = T::PriceRegistry::get(&self.info.price_id, &pool_id)?.0;
+		let delta = if amount.settlement_price > price {
+			amount.settlement_price.ensure_sub(price)?
+		} else {
+			price.ensure_sub(amount.settlement_price)?
+		};
+		let variation = T::PerThing::from_rational(delta.ensure_into()?, price.ensure_into()?);
```

Comment:
This is not correct IMO. 

```rust
let th = Perquintill::from_rational(100u64, 1u64);
assert_eq!(th, Perquintill::one())
```
If `max_price_variation` is `1`
* you can easily create a $\Delta > price$ 
* but the variation will saturate at `1`

I think we must use a `FixedPointNumber` here.

Reply:
Good catch! it saturates. We still want a `PerThing`, but we need to return an error in case of saturating here.

Reply:
Fixed now!

Reply:
BTW, not sure why there is not a `try_from_rational` or so